doc/source/development/rfc/rfc104_gdal_cli.rst

```diff
@@ -0,0 +1,1175 @@
+  lower-case capitalized: ``--long-name``
+
+  When a single value is expected, it must be separated with a space or equal sign:
+  ``--long-name <value>`` or ``--long-name=<value>``.
+
+  In the rest of the document, we will use the version with a space separator,
+  but equal sign is also accepted.
+
+Repeated values / multi-valued options
+++++++++++++++++++++++++++++++++++++++
+
+Existing GDAL command line utilities have an inconsistent strategy regarding
+how to specify repeated values (band indices, nodata values, etc.), sometimes
+with the switch being repeated many times, sometimes with a single switch but
+the values being grouped together and separated with spaces or commas.
+
+With this RFC, when several values are expected (or allowed), they must be
+separated with a ``,`` (comma): ``--bbox <xmin>,<ymin>,<xmax>,<ymax>``
+
+An alternative when passing several values is also to repeat the option as
+many times as needed: ``--co KEY1=VALUE1 --co KEY2=VALUE2`` and
+``--co KEY1=VALUE1,KEY2=VALUE2``.
+
+
+Specification of input and output files/datasets
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Two possibilities will be offered:
+
+* positional arguments with input(s) first, output last
+
+  .. code-block:: shell
+
+        gdal <subcommand> <input1> [<input2>]... <output>
+
+* using ``-i / --input`` and ``-o / --output``
+
+  .. code-block:: shell
+
+        gdal <subcommand> -i <input1> [-i <input2>]... -o <output>
+
+Reserved switches
++++++++++++++++++
+
+The following switches are reserved. Meaning that if a subcommand uses them,
+it must be with their below semantics and syntax.
+
+* ``-h``, ``--help``: display detailed help synopsis
+
+* ``-i <name>``, ``--input <name>``: specify input file/dataset
+
+* ``-o <name>``, ``--output <name>``: specify output file/dataset
+
+* ``--overwrite``: whether overwriting the output file is allowed. Defaults to no, that is execution will fail if the output file already exists.
+
+* ``-f <format>``, ``--of <format>``: output format. Value is a (not always so)
+  "short" driver name: ``GTiff``, ``COG``, ``GPKG``, ``ESRI Shapefile``.
+  Also used by ``gdal info`` to select JSON vs text output
+
+* ``--if <format>``: input format. Value is a short driver name.
+  Used when autodetection of the appropriate driver fails.
+
+* ``-b <band_number>``, ``--band <band_number>``: specify input raster band number.
+  May be repeated for utilities supporting multiple bands
+
+* ``-l <name>``, ``--layer <name>``: specify input vector layer name.
+  May be repeated for utilities supporting multiple layers
+
+* ``--co <NAME>=<VALUE>``: driver specific creation option. May be repeated.
+
+* ``--oo <NAME>=<VALUE>``: driver specific open option. May be repeated.
+
+* ``--ot {Byte|UInt16|...}``: output data type (for raster output)
+
+* ``--bbox <xmin> <ymin> <xmax> <ymax>``: as used by ``gdal vector info``,
+  ``gdal vector convert``, ``gdal convert raster``
```

Comment:
Should it be with commas now as on line 535? `--bbox <xmin>,<ymin>,<xmax>,<ymax>`

Reply:
indeed, thanks. fixed